checkdeps.py

```diff
@@ -1,4 +1,13 @@
-"""Check dependendies and give recommendations about how to satisfy them"""
+"""
+Check dependendies and give recommendations about how to satisfy them
+
+Limitations:
+
+    * Does not detect whether packages are already installed. Solving this requires writing more of a configuration
+    management system. Or we could switch to an existing one.
+    * Not fully PEP508 compliant. Not slightly. It makes bold assumptions about the simplicity of the contents of
+    EXTRAS_REQUIRE. This is fine because most developers do, too.
+"""
 
 from distutils.errors import CompileError
 try:
```

Comment:
I'll catch all that when I do the code style and linting changes. Since we're progressing through doing this to the files as a parallel task to this fix it would be unwise to conflate the two very distinct types of work.